source/JsonObject.cpp

```diff
@@ -16,26 +16,24 @@ namespace Aws
     {
         JsonObject::JsonObject() : m_wasParseSuccessful(true) { m_value = nullptr; }
```

Comment:
prior art: m_value is NULL, but m_wasParseSuccessful is true. .... that's misleading

unfortunately, this JsonObject class has a concept of "exists but invalid"

but with our C API, there's no such thing as an invalid aws_json_value. Either it's NULL, or it exists and is valid

So ... we need to square that circle. But we can't break this C++ API, right? Are IoT customers using these Json classes directly?

Reply:
I am not sure on if anyone is using this API directly or not on the IoT side. I do think we do not want to break the API if we can help it, but on the other hand, I agree that it is misleading and probably should be fixed...

I will investigate and see if I can find usage of this API to determine if we can change it.

Reply:
I just got rid of the `m_parseSuccessful` and `m_errorMessage` members. Now copying is much easier.

Now the class has exactly 1 member, and if it's nullptr then the parse was obviously not successful, and there's only 1 possible error messsage which is "Failed to parse JSON"